CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc

```diff
@@ -79,10 +81,19 @@ SiPixelQualityESProducer::SiPixelQualityESProducer(const edm::ParameterSet& conf
   auto label =
       conf_.exists("siPixelQualityLabel") ? conf_.getParameter<std::string>("siPixelQualityLabel") : std::string{};
 
-  if (label == "forDigitizer" || label == "forRawToDigi") {
+  if (label == "forDigitizer") {
     labelTokens_ =
         Tokens(setWhatProduced(this, &SiPixelQualityESProducer::produceWithLabel, edm::es::Label(label)), label);
   }
+
+  label = conf_.exists("siPixelQualityLabel_RawToDigi")
```

Comment:
`exists` is a deprecated API. The replacement is the use of `fillDescriptions`.

Reply:
Sure, I was actually surprised that there was not even an empty `fillDescriptions` but then I thought that rules were different for EDPrducers and ESProducers. So thank you for pointing out that `fillDescriptions` should be used. However, I would leave this for a follow-up PR as this one deals with a very specific problem and, as far as I understood, needs to be backported to 12_4_X rather urgently. Any additions to this PR will just cause further delays.

Reply:
Well, not really sure, I haven't really dealt with `fillDescriptions` in a while. For a complete implementation one also needs to switch to an automatically generated cfi file which potentially leads to some knock-on effects in other parts of the code. In any case, this is likely at least 1 day of delay assuming I get to it tomorrow. On the other hand, I really don't see much point in backporting `fillDescriptions` to already closed release cycles. However, since I already have a working area set up, it would be a pity not to implement `fillDescriptions` for the currently open release cycle. I would just do it in a separate PR.

Reply:
@ferencek ok, from my (`alca`) side, we can then have the backports to 12_4_X and 12_5_X identical to this PR, and later a followup PR to add `fillDescriptions` in master only.

@cms-sw/orp-l2 @Dr15Jones do you think this plan is acceptable?

Reply:
@ferencek while we wait for feedback from you and release managers on the plan I detailed detailed above I will open tha backports of this PR based on what is implemented now. I hope that is fine with you and everyone!

Reply:
@francescobrivio, the backport branches are already in place (https://github.com/CMSTrackerDPG/cmssw/tree/SiPixelQualityESProducer_fix-12_5_X and https://github.com/CMSTrackerDPG/cmssw/tree/SiPixelQualityESProducer_fix-12_4_X). Feel free to make PRs.Thanks